update.sh

```
@@ -0,0 +1,140 @@
+# ── 4. verify, else rollback ─────────────────────────────────
+log "Waiting for backend health…"
+if wait_healthy codestar-backend 60; then
+  ok "Update applied successfully → $(git rev-parse --short HEAD)"
+  docker image prune -f >/dev/null 2>&1 || true
+  exit 0
+fi
+
+err "Backend unhealthy after update. Rolling back to $(git rev-parse --short "$OLD_COMMIT")…"
+git reset --hard "$OLD_COMMIT"
+$COMPOSE up --build -d
+[ -n "$BACKUP_FILE" ] && restore_db "$BACKUP_FILE" || warn "No DB restore (backup was skipped)."
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)